codex-ui/src/vue/components/alert/core/types.ts

```diff
@@ -0,0 +1,138 @@
+import type { App, Ref } from 'vue';
+
+export enum AlertType {
+  SUCCESS = 'success',
+  ERROR = 'error',
+  WARNING = 'warning',
+  INFO = 'info',
+  DEFAULT = 'default'
+}
+
+export enum EVENTS {
+  ADD = 'add',
+  DISMISS = 'dismiss',
+  UPDATE = 'update',
+  CLEAR = 'clear'
+}
+
+export enum POSITION {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  TOP_LEFT = 'top-left',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  TOP_CENTER = 'top-center',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  TOP_RIGHT = 'top-right',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  BOTTOM_LEFT = 'bottom-left',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  BOTTOM_CENTER = 'bottom-center',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  BOTTOM_RIGHT = 'bottom-right'
+}
+
+export declare type ALERTID = string | number;
+
+export declare type Icon = string;
+
+interface AlertOptions {
+  /**
+   *  ID of the alert.
+   */
+  id?: ALERTID;
+  /**
+   * Custom icon class to be used.
+   *
+   * When set to `true`, the icon is set automatically depending on the toast type and `false` disables the icon.
+   */
+  icon?: Icon;
+  /**
+   * content to be rendered
+   */
+  content: string;
+  /**
+   *  Type of the toast.
+   *
+   *  Can be any of `(success, error, default, info, warning)`
+   */
+  status: AlertType;
+  /**
+   *  Position of the toast on the screen.
+   *
+   *  Can be any of (top-right, top-center, top-left, bottom-right, bottom-center, bottom-left).
+   */
+  position: POSITION;
```

Comment:
by design, alerts should be always at the same position, so this feature is redundant